unittests/compiled_tests.cpp

```diff
@@ -352,6 +352,11 @@ TEST_CASE("Functor cast")
   CHECK(d == 3 * 6);
 }
 
+TEST_CASE("Non-ASCII characters in string")
+{
+  chaiscript::ChaiScript_Basic chai(create_chaiscript_stdlib(),create_chaiscript_parser());
+  CHECK_THROWS_AS(chai.eval<std::string>("prin\xeft \"Hello World\""), chaiscript::exception::eval_error);
```

Comment:
You misunderstood me. I think there should be three test cases, I could even imagine five:

* Read and run a UTF-8 BOM file – no exception, normal execution
* Run a string with UTF-8 BOM – exception? could need others' opinion on this one I guess
* Run a string with an invalid character at the start – exception
* Run a string with an invalid character in the middle – exception
* Run a string with an invalid character at the end – exception